-export-dynamic is a linker option, pass it down as such.
http://git.savannah.gnu.org/cgit/gforth.git/commit/?id=9631de152b4c83cdd0aadeb179a0c4f731192174

--- configure.orig	Tue Jan 26 09:44:37 2016
+++ configure	Tue Jan 26 09:44:50 2016
@@ -4724,7 +4724,7 @@ $as_echo "$make_type" >&6; }
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the linker accepts -export-dynamic" >&5
 $as_echo_n "checking whether the linker accepts -export-dynamic... " >&6; }
 OLDLDFLAGS=$LDFLAGS
-LDFLAGS="$LDFLAGS -export-dynamic"
+LDFLAGS="$LDFLAGS -Wl,-export-dynamic"
 if test "$cross_compiling" = yes; then :
   ac_export_dynamic=no
 else
